boot/flags.go

```diff
@@ -296,7 +297,7 @@ func setNextBootFlags(dev snap.Device, rootDir string, flags []string) error {
 // ubuntu-data in an untrusted manner, but for the purposes of this function
 // that is ignored.
 // This is primarily meant to be consumed by "snap{,ctl} system-mode".
-func HostUbuntuDataForMode(mode string) ([]string, error) {
+func HostUbuntuDataForMode(mode string, mod gadget.Model) ([]string, error) {
```

Comment:
looking at the changes you had to make it seems it could be made directly to take a snap.Device, instead of separate mode and model

Reply:
Happy to change it to take a `snap.Device` instead of `gadget.Model`. But unless I'm missing something we still need the "mode" because `snap.Device` only carries `RunMode() bool` but not the full mode string?

Reply:
my confusion, we added SystemMode to DeviceContext but not to Device. We can probably adjust this. But sounds a follow up. Can you at least leave a TODO here?